src/scenario.rs

```diff
@@ -0,0 +1,548 @@
+			ScenarioExecutor::Eth(mut inner) => {
+				let (_, logs) =
+					futures::future::join(inner.resubmission_queue, inner.runner.run()).await;
+				logs
+			},
+			ScenarioExecutor::Substrate(mut inner) => {
+				let (_, logs) =
+					futures::future::join(inner.resubmission_queue, inner.runner.run()).await;
+				logs
+			},
+		}
+	}
+
+	/// Installs a ctrl_c handler which sends a stop notification on the executor
+	/// stop sender channel, to notify the stop of the scenario for displaying partial stats about
+	/// the transactions execution.
+	///
+	/// Can be called only once for the lifetime of a the program execution.
+	fn install_ctrlc_stop_hook(&self) {
+		let stop_sender = match &self {
+			ScenarioExecutor::Eth(inner) => inner.stop_sender.clone(),
+			ScenarioExecutor::Substrate(inner) => inner.stop_sender.clone(),
+		};
+		ctrlc::set_handler(move || {
+			block_on(stop_sender.send(())).expect("Could not send signal on channel.")
+		})
+		.expect("Error setting Ctrl-C handler");
+	}
+}
+/// Building logic for the execution of a scenario.
+pub struct ScenarioBuilder {
+	start_id: Option<String>,
+	last_id: Option<u32>,
+	nonce_from: Option<u128>,
+	txs_count: u32,
+	tx_recipe: Option<TransactionRecipe>,
+	does_block_monitoring: bool,
+	watched_txs: bool,
+	send_threshold: Option<usize>,
+	rpc_uri: Option<String>,
+	chain_type: Option<ChainType>,
+	installs_ctrl_c_stop_hook: bool,
+}
+
+impl ScenarioBuilder {
+	/// A default initializer of the builder, with a few defaults:
+	/// - `tx_recipe` is set to [`TransactionCall::Transfer`]
+	/// - `does_block_monitoring` is set to `false`.
+	/// - `installs_ctrl_c_stop_hook` is set to `false`.
+	/// - `send_threshold` is set to `1000`.
+	pub fn new() -> Self {
+		ScenarioBuilder {
+			start_id: None,
+			last_id: None,
+			nonce_from: None,
+			txs_count: 1,
+			tx_recipe: Some(TransactionRecipe::transfer()),
+			does_block_monitoring: false,
+			watched_txs: false,
+			send_threshold: Some(1000),
+			rpc_uri: None,
+			chain_type: None,
+			installs_ctrl_c_stop_hook: false,
+		}
+	}
+
+	/// Configure the account that represents the first signer used for the transactions
+	/// building. If the account is representing a number, and the builder isn't configured with a
+	/// last account to sign the last batch of transactions planned for execution, then the scenario
+	/// builder will build transactions only for the account specified with this setter.
+	///
+	/// The parameter is a string that can represent a number or a default development account
+	/// derivation path (e.g. "alice", "bob" etc.). If a number is given (e.g. "0", "1", "2"), it
+	/// will be interpreted as the last part of a default derivation path (see
+	/// [`subxt_transaction::derive_accounts`]), used to generate signer accounts, which are assumed
+	/// to be initialised as development accounts with balances by the networks where the
+	/// transactions will be executed.
+	pub fn with_start_id(mut self, start_id: String) -> Self {
```

Comment:
Could we easily have `u32` param here?
Maybe we could add new function `with_account_id` for passing strings?

 `with_start_id("0").with_last_id(99)` is not intuitive.

Reply:
Yes, I agree it is not intuitive. I added `with_account_id` now, and the API looks cleaner.